qutip/solver/steadystate.py

```diff
@@ -268,6 +278,28 @@ def _steadystate_svd(L, **kw):
     return rho / rho.tr()
 
 
+def _steadystate_expm(L, rho=None, propagator_tol=1e-5, **kw):
+    if rho is None:
+        from qutip import rand_dm
+        rho = rand_dm(L.dims[0][0])
+    # Propagator at an arbitrary long time
+    prop = (100 * L).expm()
+
+    rho_prev = qzero_like(rho)
+    niter = 0
+    while hilbert_dist(rho, rho_prev) > propagator_tol and niter < 30:
+        rho_prev = rho
+        rho = prop(rho)
+        prop = prop @ prop
+        rho = (rho + rho.dag()) / (2 * rho.tr())
+        niter += 1
+
+    if niter == 30:
+        raise Exception("Did not converge to a steadystate.")
```

Comment:
It might be true that the distance is already less than `propagator_tol` here and we raise an unnecessary error.